test/functional/test-fixed-layer.js

```diff
@@ -1512,6 +1537,24 @@ describes.sandboxed('FixedLayer', {}, () => {
       expect(layer.getAttribute('test1')).to.equal(null);
       expect(layer.getAttribute('test2')).to.equal(null);
     });
+
+    it('should sync invalid-named attributes to layer', () => {
+      // Holy poop it's hard to inject an invalid-named attribute.
+      let div = document.createElement('div');
+      div.innerHTML = '<div [class]="amp-bind"></div>';
+      div = div.firstElementChild;
+      const attr = div.attributes[0];
+      expect(attr.name).to.equal('[class]');
+      expect(attr.value).to.equal('amp-bind');
+
+      const body = document.createElement('body');
+      body.attributes.setNamedItem(attr.cloneNode(false));
+
+      expect(div.getAttribute('[class]')).to.equal('amp-bind');
+      expect(body.getAttribute('[class]')).to.equal('amp-bind');
+      expect(body.attributes.getNamedItem('[class]')).to.not.equal(
+          div.attributes.getNamedItem('[class]'));
```

Comment:
Shouldn't we be calling `fixedLayer.update()` and testing `layer.getAttribute(...)`?

Reply:
This is a sanity check test, it's too hard to wire a real DOM node through fixed layer (it'll take a complete refactor of this test file).

Reply:
We should at least ensure that `setNamedItem` is called. What if you mimic browser behavior by throwing on illegal chars in your fake `setAttribute()`?